xbmc/utils/FileUtils.cpp

```diff
@@ -227,4 +227,5 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   {
     CLog::Log(LOGERROR, "%s unable to extract modification date for file (%s)", __FUNCTION__, strFileNameAndPath.c_str());
   }
+  return dateAdded;
```

Comment:
Compiler is right. If an exception is thrown, we land in the catch block, which does not return or (re)throw and therefore going after that block ... and then we would return 'nothing'.